image_proc/src/nodelets/crop_decimate.cpp

```diff
@@ -195,8 +195,24 @@ void CropDecimateNodelet::imageCb(const sensor_msgs::ImageConstPtr& image_msg,
     config.height &= ~0x1;    
   }
 
+  // TODO(lucasw) need to incorporate the x_offset and y_offset within info_msg
+
+  // TODO(lucasw) should generate a black image where there isn't any source image,
+  // could be a dynamic reconfigure option
   int max_width = image_msg->width - config.x_offset;
+  if (max_width <= 0)
+  {
+    ROS_DEBUG_STREAM("x offset is outside the input image width: "
```

Comment:
I think these should be warnings, not debugs. If this is happening there's something very wrong somewhere in the pipeline and it should be made aware to a user

Reply:
The x offset can come straight from a dynamic reconfigure gui which doesn't respect the max width of the image- a flood of warnings isn't ideal when trying to dial in a number close to the edge of the true image width but somewhere in the middle of the slider range.  (Dynamic dynamic reconfigure would be nice there, set the max width and height and offsets based on the actual image instead of the cfg values)

Reply:
Having a debug statement seems like it may make it harder to fix an issue when the image itself is the issue (changing cameras, widths are different, things are silently breaking) but that's a good point during the configuration process.

Maybe a `ROS_WARN_ONCE` or `THROTTLE`? Just something on screen to help someone see something's wrong

Reply:
Throttling to once a second sounds fine, though even better may be to always warn when the max_width/height first goes negative, but then not warn again until it has gone positive and back to negative again- a couple of booleans would be used for that.

Maybe a zero size image should still be published instead of nothing at all?  Not sure if downstream nodes will handle those properly.

Reply:
Throttling or your suggestion sounds fine. I would recommend to just keep it simple and throttle every 60 seconds. If it looks like its going to be a problem someone can open a ticket to reduce spam.